snapcraft/_store.py

```diff
@@ -100,6 +101,119 @@ def _requires_login():
         raise
 
 
+@contextmanager
+def _requires_command_success(command, not_found_err=None, failure_err=None):
```

Comment:
Thanks for the pointer @elopio.
They could possibly be grouped into integration_utils.py (suggestion) but I am not sure about the gains at this point.
A small observation about the 'package' term being overloaded there. Joe's function should probably be called _check_command_exists() (according to `which` usage), it's a command/bin that is checked, not necessarily a package.